conans/client/generators/cmake_paths.py

```diff
@@ -25,9 +25,9 @@ def content(self):
         #    if the user used the "cmake_find_package" will find the auto-generated
         # 4. The CMake installation dir/Modules ones.
         deps = DepsCppCmake(self.deps_build_info)
-        lines.append("set(CMAKE_MODULE_PATH {deps.build_paths} ${{CMAKE_MODULE_PATH}} "
+        lines.append("list(APPEND CMAKE_MODULE_PATH {deps.build_paths} ${{CMAKE_MODULE_PATH}} "
                      "${{CMAKE_CURRENT_LIST_DIR}})".format(deps=deps))
```

Comment:
This change is not equivalent.
- The paths that were in `CMAKE_MODULE_PATH` will be 2x in the final `CMAKE_MODULE_PATH`.
- `deps.build_paths`, which might have higher priority will appear later in the module search path

Reply:
About your second concern: [`CMAKE_MODULE_PATH`](https://cmake.org/cmake/help/latest/variable/CMAKE_MODULE_PATH.html) is empty by default, don't you think that if the user pass it to the command line, it should have higher priority?

Reply:
You're right. I've realized now that existing `set` should be already fine because it effectively prepends Conan paths to existing ones. Strangely, this did not work properly on my machine. I need to dig into this more.